pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -180,42 +139,39 @@ type CgroupSubsystems struct {
 // cgroupManagerImpl implements the CgroupManager interface.
 // Its a stateless object which can be used to
 // update,create or delete any number of cgroups
-// It uses the Libcontainer raw fs cgroup manager for cgroup management.
+// It relies on runc/libcontainer cgroup managers.
 type cgroupManagerImpl struct {
 	// subsystems holds information about all the
 	// mounted cgroup subsystems on the node
 	subsystems *CgroupSubsystems
-	// simplifies interaction with libcontainer and its cgroup managers
-	adapter *libcontainerAdapter
+
+	// useSystemd tells if systemd cgroup manager should be used.
+	useSystemd bool
 }
 
 // Make sure that cgroupManagerImpl implements the CgroupManager interface
 var _ CgroupManager = &cgroupManagerImpl{}
 
 // NewCgroupManager is a factory method that returns a CgroupManager
-func NewCgroupManager(cs *CgroupSubsystems, cgroupDriver string) CgroupManager {
-	managerType := libcontainerCgroupfs
-	if cgroupDriver == string(libcontainerSystemd) {
-		managerType = libcontainerSystemd
-	}
+func NewCgroupManager(cs *CgroupSubsystems, useSystemd bool) CgroupManager {
```

Comment:
I think it's a bit confusing to change the type signature of this function here... while the compiler is going to ensure we don't miss any conversions, I think this might be confusing when trying to compare old vs. new code. If it's possible to lift this out of sight and avoid the change I'd do that, e.g. 

```suggestion
func NewCgroupManager(cs *CgroupSubsystems, cgroupDriver string) CgroupManager {
    useSystemd := cgroupDriver == "systemd"
```

We may also want to factor out the constant for the systemd cgroup driver name.

Reply:
I have changed that because the old interface allows an arbitrary string and won't catch the error of providing some arbitrary value.

Changed back to string.